oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/incrementalstore/IncrementalStoreOperand.java

```diff
@@ -18,10 +18,44 @@
  */
 package org.apache.jackrabbit.oak.index.indexer.document.incrementalstore;
 
+/**
+ * The operation to perform. Operations are generated by either running a 'diff'
+ * , or by 'top up' which is downloading the changes from MongoDB.
+ *
+ * When using the 'diff' command, we expect that the node doesn't exist yet for
+ * the 'add' case, and that the node exists for the 'delete' and 'modify' case
+ * (due to the nature of the 'diff'). The operations are then ADD, DELETE,
+ * MODIFY.
+ *
+ * When using the 'top up', we don't know whether the node existed before or
+ * not; we only get the updated state. That's why we use different operations:
+ * REMOVE_IF_EXISTS and INSERT_OR_UPDATE.
+ */
 public enum IncrementalStoreOperand {
+
+    // add a new node;
+    // log a warning if it doesn't exist
+    // (this is used by the 'diff' command)
     ADD("A"),
+
+    // delete an existing node;
+    // log a warning if it doesn't exist
+    // (this is used by the 'diff' command)
+    DELETE("D"),
+
+    // modify an existing node;
+    // log a warning if it doesn't exist
+    // (this is used by the 'diff' command)
     MODIFY("M"),
-    DELETE("D");
+
+    // remove a node that may or may not exist
+    // (this operation may be used by the 'top up' command)
+    REMOVE_IF_EXISTS("R"),
```

Comment:
The enum is REMOVE_IF_EXISTS but the description says remove a node that may or may not exist - should it not be - 

Remove a node if exists otherwise NOOP ? 

Also, I am not sure if it's clear to me how does it differ from DELETE ? That is also deleting and if the node is not present - then logging a warning.

Reply:
Fixed.

> how does it differ from DELETE

It differs: DELETE logs a warning if the node doesn't exist, REMOVE_IF_EXISTS does not.

Reply:
But do we really need REMOVE_IF_EXISTS then - Is it just to avoid the warning ?